datafusion/physical-plan/src/aggregates/mod.rs

```diff
@@ -817,7 +818,16 @@ fn create_schema(
         | AggregateMode::SinglePartitioned => {
             // in final mode, the field with the final result of the accumulator
             for expr in aggr_expr {
-                fields.push(expr.field()?)
```

Comment:
I think it would be better to update the defintion of `field` here to return the correct value

What I suggest is to add a `field` method to `AggregateUDFImpl` with a default implementation  like [this](https://github.com/apache/datafusion/blob/e088945c38b74bb1d86dcbb88a69dfc21d59e375/datafusion/physical-expr-functions-aggregate/src/aggregate.rs#L243-L245)

Then have `AggregateFunctionExpr::field` call into `fun.field()`

Note that the `AggregateExpr` trait is under consideration for removal -- see https://github.com/apache/datafusion/issues/11810

Reply:
Thanks. Let me improve this later.